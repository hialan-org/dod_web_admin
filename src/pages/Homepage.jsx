import React, {useEffect, useState} from 'react';
import {EMAIL} from "../constants/data";
import {getListUser, login} from "../utils/APIUtils";

function Homepage(props) {
    const [email, setEmail] = useState("");

    useEffect(() => {
        if(props.authenticated){
            setEmail(sessionStorage.getItem(EMAIL))
        }
    }, []);

    return (
        <div style={styles.screen}>
            <div style={styles.body}>
                <h1>Dog of The Dow Management</h1>
                {
                    props.authenticated ?
                        (
                            <>
                                <h2 style={styles.text}>User: {email}</h2>
                                <a href="#" onClick={props.logout}>Logout</a>
                            </>
                        ) : ""
                }
            </div>

        </div>
    );
}

const styles = {
    screen : {
        display: "flex",
        height: '100vh',
        backgroundColor:"rgb(53,58,63)"
    },
    body : {
        flex:1,
        color: 'snow',

    }
};

export {Homepage};