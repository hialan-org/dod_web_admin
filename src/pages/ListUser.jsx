import React, {useState, useEffect} from "react";
import Table from "react-bootstrap/Table";
import {deleteUser, getListUser} from "../utils/APIUtils";
import {ROLE, ROLE_ADMIN, EMAIL} from "../constants/data";
import {Redirect} from "react-router-dom";

import 'react-bootstrap-table-next/dist/react-bootstrap-table2.min.css';
import BootstrapTable from 'react-bootstrap-table-next';
import overlayFactory from 'react-bootstrap-table2-overlay';
import Spinner from "react-bootstrap/Spinner";
import Button from "react-bootstrap/Button";
import Badge from "react-bootstrap/Badge";

function ListUser(props) {
    const [users, setUsers] = useState([]);
    const [redirect, setRedirect] = useState("");
    const [isLoading, setIsLoading] = useState(true);

    useEffect(() => {
        if ((sessionStorage.getItem(ROLE) && sessionStorage.getItem(ROLE) !== ROLE_ADMIN)
            || !props.authenticated) {
            setRedirect("/");
        } else {
            getListUser().then(result => {
                setUsers(result.body);
                setIsLoading(false);
            }).catch(err => {
                console.log(err);
            })
        }
    }, []);

    function onClickDelete(userId) {
        deleteUser(userId).then(result => {
            console.log("Delete success");
            getListUser().then(result => {
                setUsers(result.body);
                setIsLoading(false);
            }).catch(err => {
                console.log(err);
            })
        }).catch(err => {
            console.log(err);
        })
    }

    function buttonFormatter(cell, row, rowIndex, formatExtraData) {
        return (
            <Button
                disabled={row.email === sessionStorage.getItem(EMAIL)}
                variant="danger"
                onClick={() => onClickDelete(row.userId)}>
                Delete
            </Button>
        );
    }

    function dateFormatter(cell, row, rowIndex, formatExtraData) {
        const date = new Date(cell);
        return (
            <span>{date.toDateString()}</span>
        )
    }

    const columns = [{
        dataField: 'userId',
        text: 'User ID'
    }, {
        dataField: 'email',
        text: 'Email'
    }, {
        dataField: 'createdDate',
        text: 'Created Date',
        formatter: dateFormatter,
    }, {
        dataField: 'role',
        text: 'Role',
    }, {
        dataField: "actions",
        text: "Actions",
        sort: false,
        formatter: buttonFormatter,
        headerAttrs: {width: 100},
    }];

    return (
        redirect ?
            (<Redirect push to="/"/>)
            : (
                <>
                    <div className="list-users-header">
                            <Badge pill variant="primary">Total Users: {users.length}</Badge>
                            <Badge pill variant="secondary">Total Users's Stock: </Badge>
                            <Badge pill variant="secondary">Total Amount of User Money: </Badge>
                    </div>
                    <BootstrapTable
                        keyField='userId'
                        data={users}
                        columns={columns}
                        noDataIndication={() => <Spinner animation="border"/>}
                    />
                </>
            )
    )
}

export {ListUser}