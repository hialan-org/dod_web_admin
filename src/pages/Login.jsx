import React, {useState} from 'react';
import { Redirect } from "react-router-dom";
import {GoogleLoginButton} from "../components/GoogleLoginButton";
import {ACCESS_TOKEN, EMAIL, ROLE} from "../constants/data";
import {login} from "../utils/APIUtils";

function Login(props) {
    return (
        props.authenticated ?  (
            <Redirect push to="/"/>
        ) : (
            <div className="App">
                <GoogleLoginButton onClick={props.responseGoogle}/>
            </div>
        )
    );
}

export {Login}