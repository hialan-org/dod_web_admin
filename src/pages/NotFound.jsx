import React from 'react';
import {Link} from "react-router-dom";

function NotFound(props) {
    return (
        <div>
            <img src="https://i.imgur.com/6F707L7.jpg" title="source: imgur.com" style={{height: 600, display: 'block', margin: 'auto', position: 'relative' }}/>
        </div>
    )
}

export {NotFound}