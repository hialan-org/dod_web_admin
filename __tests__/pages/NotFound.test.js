import React from 'react';
import renderer from 'react-test-renderer';
import {NotFound} from "../../src/pages/NotFound";

const {act} = renderer;

describe('<NotFound />', () => {
    it('test match snapshot', async () => {
        let notFound;
        await act(async () => {notFound = renderer.create(
            <NotFound
            />)
        });
        expect(notFound.toJSON()).toMatchSnapshot();

    });
})