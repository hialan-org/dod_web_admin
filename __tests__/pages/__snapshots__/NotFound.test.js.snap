// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound /> test match snapshot 1`] = `
<div>
  <img
    src="https://i.imgur.com/6F707L7.jpg"
    style={
      Object {
        "display": "block",
        "height": 600,
        "margin": "auto",
        "position": "relative",
      }
    }
    title="source: imgur.com"
  />
</div>
`;
